Sources/Datadog/DatadogInternal/Codable/AnyEncoder.swift

```diff
@@ -0,0 +1,651 @@
+        /// - parameter value: The value to encode.
+        func encode(_ value: Float) throws {
+            storage.any = value
+        }
+
+        /// Encodes a single value of the given type.
+        ///
+        /// - parameter value: The value to encode.
+        func encode(_ value: Int) throws {
+            storage.any = value
+        }
+
+        /// Encodes a single value of the given type.
+        ///
+        /// - parameter value: The value to encode.
+        func encode(_ value: Int8) throws {
+            storage.any = value
+        }
+
+        /// Encodes a single value of the given type.
+        ///
+        /// - parameter value: The value to encode.
+        func encode(_ value: Int16) throws {
+            storage.any = value
+        }
+
+        /// Encodes a single value of the given type.
+        ///
+        /// - parameter value: The value to encode.
+        func encode(_ value: Int32) throws {
+            storage.any = value
+        }
+
+        /// Encodes a single value of the given type.
+        ///
+        /// - parameter value: The value to encode.
+        func encode(_ value: Int64) throws {
+            storage.any = value
+        }
+
+        /// Encodes a single value of the given type.
+        ///
+        /// - parameter value: The value to encode.
+        func encode(_ value: UInt) throws {
+            storage.any = value
+        }
+
+        /// Encodes a single value of the given type.
+        ///
+        /// - parameter value: The value to encode.
+        func encode(_ value: UInt8) throws {
+            storage.any = value
+        }
+
+        /// Encodes a single value of the given type.
+        ///
+        /// - parameter value: The value to encode.
+        func encode(_ value: UInt16) throws {
+            storage.any = value
+        }
+
+        /// Encodes a single value of the given type.
+        ///
+        /// - parameter value: The value to encode.
+        func encode(_ value: UInt32) throws {
+            storage.any = value
+        }
+
+        /// Encodes a single value of the given type.
+        ///
+        /// - parameter value: The value to encode.
+        func encode(_ value: UInt64) throws {
+            storage.any = value
+        }
+
+        /// Encodes a single value of the given type.
+        ///
+        /// - parameter value: The value to encode.
+        func encode<T>(_ value: T) throws where T: Encodable {
+            if value is SharedEncodable {
+                storage.any = value
+            } else {
+                let encoder = _AnyEncoder(path: codingPath)
+                try value.encode(to: encoder)
+                storage.any = encoder.storage?.any
+            }
+        }
+    }
+}
+
+/// A shared encodable object will skip encoding when using the `AnyEncoder`.
+///
+/// Making a Encodable as shared allow to bypass encoding when the type is
+/// known by multiple parties.
+internal protocol SharedEncodable: Encodable { }
```

Comment:
I think we could consider `(Shared)DictionaryEncodable`. I think it would be puzzling to developers to custom encodable, until they realised it's shortening the path and allows encoding directly to a Dictionary.